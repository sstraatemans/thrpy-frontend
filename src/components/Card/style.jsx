import styled from "styled-components";
import { H4 } from "./../../style";

export const CardContainer = styled.div`
  position: relative;
  background-color: #fff;
  box-shadow: ${props => props.theme.boxShadow}
  border-radius: 2px;

  text-align: left;
`;

export const CardContent = styled.div`
  padding: 24px;
  border-radius: 0 0 2px 2px;
`;

export const CardTitle = styled.h4`
  display: block;
  line-height: 32px;
  margin-bottom: 8px;
  margin-top: 0;
  font-size: 24px;
  font-weight: 300;
`;
