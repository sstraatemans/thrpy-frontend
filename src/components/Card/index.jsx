import React from "react";
import { CardContainer, CardContent, CardTitle } from "./style";
type Props = {
  children: any,
  title?: string
};
const Card = (props: Props) => {
  const getTitle = () => {
    if (props.title) {
      return <CardTitle>{props.title}</CardTitle>;
    }
  };
  return (
    <CardContainer>
      <CardContent>
        {getTitle()}
        <p>{props.children}</p>
      </CardContent>
    </CardContainer>
  );
};

export default Card;
