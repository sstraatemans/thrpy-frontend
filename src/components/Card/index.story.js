import React from "react";

import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";
import Card from "./";

storiesOf("Card", module)
  .add("no title", () => <Card />)
  .add("with title", () => <Card title="Title goes here" />);
