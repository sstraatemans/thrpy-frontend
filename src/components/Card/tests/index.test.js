import React from "react";
import Card from "./../";

describe("<Card />", () => {
  it("render component <Card>", () => {
    let wrapper = mount(<Card title="title goes here" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
