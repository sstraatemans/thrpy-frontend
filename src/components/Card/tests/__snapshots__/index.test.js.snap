exports[`<Card /> render component <Card> 1`] = `
<Card
  title="title goes here">
  <styled.div>
    <div
      className="sc-htpNat lgJFQD">
      <styled.div>
        <div
          className="sc-bxivhb dqQpey">
          <styled.h4>
            <h4
              className="sc-ifAKCX iRjmoV">
              title goes here
            </h4>
          </styled.h4>
          <p />
        </div>
      </styled.div>
    </div>
  </styled.div>
</Card>
`;
